Cabal/Distribution/Simple.hs

```diff
@@ -149,8 +153,9 @@ defaultMainWithHooksNoReadArgs hooks pkg_descr =
   defaultMainHelper hooks { readDesc = return (Just pkg_descr) }
 
 defaultMainHelper :: UserHooks -> Args -> IO ()
-defaultMainHelper hooks args = topHandler $
-  case commandsRun (globalCommand commands) commands args of
+defaultMainHelper hooks args = topHandler $ do
```

Comment:
Since in principle a Setup script doesn't have to depend on `lib:Cabal` or use one of `defaultMain*` variants, we should mention in the documentation that setup scripts are required to implement response file support starting with spec version 3.0.